src/amo/components/AddAddonToCollection/index.js

```diff
@@ -0,0 +1,301 @@
+/* @flow */
+/* global window */
+/* eslint-disable react/sort-comp */
+import makeClassName from 'classnames';
+import React from 'react';
+import { connect } from 'react-redux';
+import { compose } from 'redux';
+
+import {
+  addAddonToCollection, fetchUserCollections,
+} from 'amo/reducers/collections';
+import { withFixedErrorHandler } from 'core/errorHandler';
+import translate from 'core/i18n/translate';
+import log from 'core/logger';
+import Select from 'ui/components/Select';
+import Notice from 'ui/components/Notice';
+import type { AddonType } from 'core/types/addons';
+import type { ErrorHandlerType } from 'core/errorHandler';
+import type { I18nType } from 'core/types/i18n';
+import type { DispatchFunc } from 'core/types/redux';
+import type {
+  CollectionId, CollectionsState, CollectionType,
+} from 'amo/reducers/collections';
+import type { UsersStateType } from 'amo/reducers/users';
+import type { ElementEvent } from 'core/types/dom';
+
+import './styles.scss';
+
+
+type Props = {|
+  addon: AddonType | null,
+  className?: string,
+  dispatch: DispatchFunc,
+  errorHandler: ErrorHandlerType,
+  i18n: I18nType,
+  loadingAddonsInCollections: boolean,
+  loadingUserCollections: boolean,
+  siteUserId: number | null,
+  // These are all user collections that the current add-on is a part of.
+  addonInCollections: Array<CollectionType> | null,
+  // These are all collections belonging to the user.
+  userCollections: Array<CollectionType> | null,
+  _window: typeof window | Object,
+|};
+
+type OnSelectOptionType = () => void;
+
+type SelectData = {| options: Array<Object>, disabled: boolean |};
+
+export class AddAddonToCollectionBase extends React.Component<Props> {
+  optionSelectHandlers: { [key: string]: OnSelectOptionType };
+
+  static defaultProps = {
+    _window: typeof window !== 'undefined' ? window : {},
+  };
+
+  constructor(props: Props) {
+    super(props);
+    this.optionSelectHandlers = {};
+  }
+
+  componentDidMount() {
+    // This runs as componentDidMount() to only load data client side,
```

Comment:
Oh. The reason it was client only was because it made some API requests which would have increased the server load. It no longer makes those API requests so I'll put it back on the server.

Reply:
Actually, nevermind. It still loads data from the collections API when it first renders. I think we should only do it client-side because otherwise it introduces more load on the server. We don't need the select box for SEO so that should be fine. It won't work with JS disabled so why bother.

Reply:
I'll add a comment to explain why

Reply:
Ah, cool, that totally makes sense. Comment is perfect, thanks 👍